pandas/core/arrays/period.py

```diff
@@ -803,8 +798,12 @@ def _add_timedelta_arraylike(
                 np.asarray(other), dtype=dtype, copy=False, round_ok=False
             )
         except ValueError as err:
-            # TODO: not actually a great exception message in this case
-            raise raise_on_incompatible(self, other) from err
+            # e.g. if we have minutes freq and try to add 30s
+            # "Cannot losslessly convert units"
+            raise IncompatibleFrequency(
```

Comment:
At some point, could you expose this exception in `pandas/errors/__init__.py` since I think the user can encounter this error?